app/src/main/java/com/jalasoft/routesapp/ui/auth/login/view/LoginFragment.kt

```diff
@@ -1,32 +1,208 @@
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Observer
 import androidx.navigation.fragment.findNavController
+import com.facebook.*
+import com.facebook.login.LoginManager
+import com.facebook.login.LoginResult
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.tasks.Task
+import com.google.firebase.auth.AuthCredential
+import com.google.firebase.auth.FacebookAuthProvider
+import com.google.firebase.auth.GoogleAuthProvider
 import com.jalasoft.routesapp.R
+import com.jalasoft.routesapp.databinding.FragmentLoginBinding
+import com.jalasoft.routesapp.ui.auth.login.viewModel.LoginViewModel
+import com.jalasoft.routesapp.util.helpers.UserTypeLogin
+import dagger.hilt.android.AndroidEntryPoint
+import org.json.JSONException
 
+@AndroidEntryPoint
 class LoginFragment : Fragment() {
 
+    private var _binding: FragmentLoginBinding? = null
+    private val binding get() = _binding!!
+    private val viewModel: LoginViewModel by viewModels()
+    private lateinit var googleSingInClient: GoogleSignInClient
+    private lateinit var callbackManager: CallbackManager
+    private lateinit var loginManager: LoginManager
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        observers()
     }
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
-        return inflater.inflate(R.layout.fragment_login, container, false)
+    ): View {
+        _binding = FragmentLoginBinding.inflate(inflater, container, false)
+        return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        view.findViewById<Button>(R.id.btn_log_register).setOnClickListener {
+        googleConfiguration()
+        facebookConfiguration()
+        viewModel.context = context
+        binding.btnLogRegister.setOnClickListener {
             findNavController().navigate(R.id.action_loginFragment_to_registerUserFragment)
         }
+
+        binding.btnLogin.setOnClickListener {
+            loginWithEmailAndPassword()
+        }
+
+        binding.ibLogGoogle.setOnClickListener {
+            signInGoogle()
+        }
+
+        binding.ibLogFacebook.setOnClickListener {
+            loginManager.logIn(this, listOf("email", "public_profile"))
+        }
+    }
+
+    @Deprecated("Deprecated in Java")
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
```

Comment:
instead of using onActivityResult, use its alternative which is registerForActivityResult.
Check this https://stackoverflow.com/questions/62671106/onactivityresult-method-is-deprecated-what-is-the-alternative

Reply:
Facebook activities (WebView and App) are still launched the old way. I couldn't find in documentation a way to launch those intents through kotlin launcher